### PR TITLE
[Java] Add vertx

### LINKED ISSUE #441: vertx java framework
+ Language : Java
+ Repo : https://vertx.io/ https://github.com/eclipse-vertx/vert.x
+ Author : https://github.com/eclipse-vertx

### PR DESCRIPTION
**Pull request by waghanza**
Hi,

This `PR` introduces vertx.

Regards,

-----------
@pmlopes @jponge Would you mind review this `PR` ?

-----------
Closes #441

### CODE REVIEW

File: java/vertx4web/src/main/java/benchmark/vertx4web/BenchmarkApplication.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import io.vertx.ext.web.Router;
+
+public class BenchmarkApplication {
+    public static void main( String [] args )
+    {
+        Vertx vertx = Vertx.vertx();
+
+        Router router = Router.router(vertx);
+
+        router.get("/").respond( ctx -> ctx.response().setStatusCode(200).end());
```
Comment (pmlopes) [inline java/vertx4web/src/main/java/benchmark/vertx4web/BenchmarkApplication.java:14]: No need to set status, it's set by default to 200.

**Review by pmlopes — CHANGES_REQUESTED** [review]
Overall this is a good start. In order to get the top performance, I'd recommend a few changes, mostly on the setup, not on the code itself.

Let me know if assistance is needed to get the changes in.

**Comment by waghanza** [comment]
Thanks for all your feedback @pmlopes. I must have told this before, the first purpose (for this project) is to reflect real-life usages. The idea is not to have the maximum performance, but a mix of :
+ Security
+ Safety
+ Scalability
+ Best-practices 
+ ...

It really depends on the community behind the framework.

**Comment by pmlopes** [comment]
Ok in that case most of the startup flags I've listed can be ignored. I'll try to make a PR on the PR tomorrow so we can discuss it further.

File: java/vertx/pom.xml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        <maven.compiler.target>11</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <build>
+        <finalName>server</finalName>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (waghanza) [reply java/vertx/pom.xml:26]: I'm not sure how to do that @pmlopes

**Comment by waghanza** [comment]
@pmlopes time to `PR` ?

**Comment by pmlopes** [comment]
Hi, I've been busy with other tasks. I hope to get it ready tomorrow or beginning next week.

**Comment by waghanza** [comment]
ok

I prefer that a professional `java` developer make this implementation 😛

**Pull request closed without merging**